src/main/java/com/cloudbees/jenkins/support/impl/SystemConfiguration.java

```diff
@@ -0,0 +1,110 @@
+package com.cloudbees.jenkins.support.impl;
+
+import com.cloudbees.jenkins.support.AsyncResultCache;
+import com.cloudbees.jenkins.support.api.*;
+import com.cloudbees.jenkins.support.util.SystemPlatform;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.model.Computer;
+import hudson.model.Node;
+import hudson.security.Permission;
+import hudson.slaves.SlaveComputer;
+import jenkins.model.Jenkins;
+
+import java.io.*;
+import java.util.*;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+
+/**
+ * System configuration data (CPU information, swap configuration, mount points,
+ * kernel messages and entropy)
+ */
+@Extension
+public class SystemConfiguration extends Component {
+
+    private final WeakHashMap<Node,SystemPlatform> systemPlatformCache = new WeakHashMap<Node, SystemPlatform>();
+
+    private final WeakHashMap<Node,String> sysCtlCache = new WeakHashMap<Node, String>();
+
+    private final WeakHashMap<Node,String> userIdCache = new WeakHashMap<Node, String>();
+
+    private static final Logger LOGGER = Logger.getLogger(SystemConfiguration.class.getName());
+
+    static Map<String,String> UNIX_PROC_CONTENTS;
+    static {
+        UNIX_PROC_CONTENTS = new HashMap<String,String>();
+        UNIX_PROC_CONTENTS.put("/proc/swaps", "swaps.txt");
+        UNIX_PROC_CONTENTS.put("/proc/cpuinfo", "cpuinfo.txt");
+        UNIX_PROC_CONTENTS.put("/proc/mounts", "mounts.txt");
+        UNIX_PROC_CONTENTS.put("/proc/sys/kernel/random/entropy_avail", "kernel_entropy_avail.txt");
```

Comment:
Probably irrelevant after the fix of [JENKINS-20108](https://issues.jenkins-ci.org/browse/JENKINS-20108).